iosHackthonApp/Modules/Features/AddMeal/Sources/AddMeal/AddMealView.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  AddMealView.swift
+//  iosHackthonApp
+//
+//  Created by Gareth Miller on 09/11/2020.
+//  Copyright © 2020 orgName. All rights reserved.
+//
+
+import SwiftUI
+import shared
+import Strings
+import Theming
+import Location
+import Components
+
+public struct AddMealView: View {
+
+    public init(viewModel: AddMealViewModel) {
+        self.viewModel = viewModel
+    }
+
+    @ObservedObject private(set) var viewModel: AddMealViewModel
+    @State var title: String = ""
+    @State var additionalInfo: String = ""
+    @State var quantity: Int = 0
+    @State var availableFromDate = Date()
+    @State var useByDate = Date()
+    @State var address: String = ""
+    @State var latitude: Float = 0
+    @State var longitude: Float = 0
+    @State var isHot: Bool = true
+
+    var buttonIsDisabled: Bool {
+        title.isEmpty || additionalInfo.isEmpty || address.isEmpty || quantity == 0
+    }
+
+    public var body: some View {
+        ZStack {
+            ScrollView {
+                VStack(alignment: .leading, spacing: 10) {
+                    Group {
```

Comment:
Could any of these things be broken out into smaller view structs to help readability? not sure how that would affect the architecture of the bindings, so fine if not

Reply:
Yeah maybe, I'll check with Raynelle in case she's tried and hit a roadblock 👍 
If not, I might attempt that as my next task separate to this if that's alright? 
I think we could probably move some of the state variables into the viewModel also

Reply:
Sounds good! I'm really new to SwiftUI, you've all probably got more experience here than me so I can only look at it from a readability perspective atm rather than architectural. Happy to take your lead here